### ISSUE TITLE
Incompatibility with Iron's Spells and Spellbooks

### ISSUE THREAD

**Issue by sentwayfarer** [Thelnfamous1/Better-Mob-Combat#14]
This is a bit of an odd and niche issue because ISS nor any of its addons utilize ISS's `WarlockAttackGoal` by default, but I've come across this issue when utilizing [KubeJS](<https://www.curseforge.com/minecraft/mc-mods/kubejs>), [KubeJS Iron's Spells](<https://www.curseforge.com/minecraft/mc-mods/kubejs-irons-spells>), and [EntityJS](<https://www.curseforge.com/minecraft/mc-mods/entityjs>). When the `WarlockAttackGoal` is active, the mob's melee attacks do not utilize Better Mob Combat functionality.

If you need to test the script I'm using for it, please go ahead. It requires all the mods listed above. It goes in a `.js` file in the `kubejs/server_scripts` folder:
```js
const $WarlockAttackGoal = Java.loadClass('io.redspace.ironsspellbooks.entity.mobs.goals.WarlockAttackGoal')

EntityJSEvents.addGoalSelectors('minecraft:husk', event => {
    event.arbitraryGoal(1, (e) => {
        return new $WarlockAttackGoal(e, 1, 40, 50, 2)
            .setMeleeBias(0.8, 0.8)
            .setAllowFleeing(false)
            .setSpells(
                [Spell.of('irons_spellbooks:firebolt'), Spell.of('irons_spellbooks:magic_missile')],
                [Spell.of('irons_spellbooks:slow')],
                [],
                []
            )
    })
})
```
https://github.com/user-attachments/assets/9642a4ec-bdae-40e1-bf50-1a7f31956cf8

**Comment by sentwayfarer** [comment]
I forgot to mention that I'm using a datapack for air punching. I'll list it here in a moment.

**Comment by sentwayfarer** [comment]
[punch.zip](https://github.com/user-attachments/files/16434139/punch.zip)
Here's the datapack.

**Comment by Thelnfamous1** [comment]
Yeah, I haven't added any special support for their `WarlockAttackGoal`. I can add it.

**Comment by sentwayfarer** [comment]
Any news on this? I couldn't fix the issue myself.
